src/harness/BenchmarkDotNet.Extensions/RecommendedConfig.cs

```diff
@@ -83,5 +92,63 @@ private static DisassemblyDiagnoser CreateDisassembler()
                 exportHtml: false,
                 exportCombinedDisassemblyReport: false,
                 exportDiff: false));
+
+        private static IEnumerable<string> GetBenchmarksToResume(DirectoryInfo artifacts)
+        {
+            if (!artifacts.Exists)
+                return new string[0];
+
+            // Get all existing report files, of any export type; order by descending filename length to avoid rename collisions
+            return artifacts.GetFiles($"*-report-*", SearchOption.AllDirectories)
+                .OrderByDescending(resultFile => resultFile.FullName.Length)
+                .SelectMany(resultFile =>
+                {
+                    var reportFileName = resultFile.FullName;
+
+                    // Prepend the report name with -resume, potentially multiple times if multiple reports for the same
+                    // benchmarks exist, so that they don't collide with one another. But don't unnecessarily prepend
+                    // -resume multiple times.
+                    if (!reportFileName.Contains("-resume-report-") || File.Exists(reportFileName.Replace("-resume-report-", "-report-")))
+                    {
+                        var resumeFileName = reportFileName.Replace("-report-", "-resume-report-");
+                        File.Move(reportFileName, resumeFileName);
+
+                        reportFileName = resumeFileName;
+                    }
+
+                    // For JSON reports, load the data to get the benchmarks that have already been reported
+                    if (reportFileName.EndsWith(".json"))
+                    {
+                        try
+                        {
+                            var result = JsonConvert.DeserializeObject<BdnResult>(File.ReadAllText(reportFileName));
+                            var benchmarks = result.Benchmarks.Select(benchmark =>
+                            {
+                                var nameParts = new[] { benchmark.Namespace, benchmark.Type, benchmark.Method };
+                                return string.Join(".", nameParts.Where(part => !string.IsNullOrEmpty(part)));
+                            }).Distinct();
+
+                            return benchmarks;
+                        }
+                        catch (JsonSerializationException)
+                        {
```

Comment:
If the last test run ended abruptly such that it left invalid JSON, it seems that we will just fail to resume. 

Another option is, if there's such an exception, grep the file for a distinctive token such as `BenchmarkDotNetVersion` to confirm it's a relevant JSOn file. If it's present, then emit the JSON exception, which will identify the corrupted file.

Reply:
What I was thinking here is that if the run ended abruptly and invalid JSON was left behind, then we will discard those completed benchmarks, allowing them to run again and produce valid JSON.